### ISSUE TITLE
Allow hiding legend in plots

### ISSUE THREAD

**Issue by mturoci** [h2oai/wave#351]
via @isarth23

Proposal: Add `legend` prop to `plot_card`, defaults to `True`.

Implementation note: `chart.legend(false)` is all it takes.

cc @lo5 @geomodular

**Comment by lo5** [comment]
I'm trying to understand the motivation behind this issue - why would you want to hide the legend on a visualization?

**Comment by mturoci** [comment]
cc @isarth23 

![image](https://user-images.githubusercontent.com/64769322/102745839-32e7be80-435d-11eb-9b35-461093c9dda8.png)

I know you consider this just a visual bug, but isn't legend in this case just unnecessary distraction? It is clear which bar represents what in this case (we should however solve the overflow).

In general, I would consider importance of the legend case by case, but providing flexibility to remove it when needed is a plus that folks might appreciate. However, I am by no means a data viz expert so I may be wrong on this one.

**Comment by lo5** [comment]
In this particular vis, color-encoding the bars is not useful. Removing the color encoding will make the legend go away.

**Comment by isarth23** [comment]
Yeah, but it seemed nice with different colors, can be added plus to include but not necessary.

**Comment by dott1718** [comment]
I have another use case where hiding the legend is important. I am marking the items that were selected via events using different color, but I don't want them to appear in the legend as 2 groups.
